### PR TITLE
fix: tokenized ECE do not show button when missing billing email

### PR DESCRIPTION
**Pull request by frosso**
Fixes #

#### Changes proposed in this Pull Request

The Store API (which is used by the Tokenized ECE to process an order) doesn't allow to process an order if the billing email is not present.
This is logged in WC Core here: https://github.com/woocommerce/woocommerce/issues/48540

However, while the discussion is happening, we'd like to quickly patch the tokenized ECE to provide a message in the merchant's logs when such thing happens.
At the same time, I'm disabling the ECE button if the billing email is not present, so that the customer doesn't see any errors when attempting to process the order.

#### Testing instructions

- As a merchant, create an order in `wp-admin`
- Ensure you're adding some information, but omit the billing address
Screenshot 2024-12-20 at 2 06 30 PM
- Save the order, and click on the "Customer payment page →" link
- The page should not display the GooglePay/ApplePay buttons
Screenshot 2024-12-20 at 2 06 20 PM
- At least one message should be added in the logs
Screenshot 2024-12-20 at 2 06 06 PM
- Update the same order to include a billing email
Screenshot 2024-12-20 at 2 06 46 PM
- Refresh the "Customer payment" page
- The ECE button should now be displayed
- You should be able to pay for the order with the ECE button
Screenshot 2024-12-20 at 2 06 53 PM

-------------------

- [x] Run `npm run changelog` to add a changelog file, choose `patch` to leave it empty if the change is not significant. You can add multiple changelog files in one PR by running this command a few times. 
- [x] Covered with tests (or have a good reason not to test in description ☝️)
- [ ] Tested on mobile (or does not apply)

**Post merge**

- [ ] Link to testing instructions from [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) following [these instructions](https://github.com/Automattic/woocommerce-payments/wiki/How-to-write-good-manual-testing-scenarios) : _Add link here / 'QA Testing Not Applicable'_
- [ ] Add or update [critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Critical-flows) and [testing instructions for critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Testing-instructions-for-critical-flows), if applicable.
- [ ] Add what's changed (description, screenshot, demo videos etc.) to the release announcement post, if applicable.

### CODE REVIEW

**Comment by botwoo** [comment]
### Test the build 

#### Option 1. Jetpack Beta

- Install and activate [Jetpack Beta](https://jetpack.com/download-jetpack-beta/).
- Use this build by searching for PR number `10019` or branch name `fix/tokenized-ece-pay-for-order-without-billing-email` in your-test.site/wp-admin/admin.php?page=jetpack-beta&plugin=woocommerce-payments          

#### Option 2. Jurassic Ninja - available for logged-in A12s

:rocket: [Launch a JN site with this branch](https://jurassic.ninja/create/?jetpack-beta&shortlived&nojetpack&woocommerce&woocommerce-payments-dev-tools&branches.woocommerce-payments=fix/tokenized-ece-pay-for-order-without-billing-email) :rocket:

:information_source: Install this [Tampermonkey script](https://github.com/Automattic/woocommerce-payments/tree/develop/bin/wcpay-live-branches) to get more options.

---

Build info: 

- Latest commit: 8848e29dd8c8ff48495b879183ea6311c9ab4906
- Build time: 2024-12-20 13:19:38 UTC

_Note: the build is updated when a new commit is pushed to this PR._

File: tests/unit/express-checkout/test-class-wc-payments-express-checkout-button-helper.php [hunk 1]
```diff
@@ -208,6 +209,30 @@ function () {
...
+		$wp->query_vars       = [ 'order-pay' => strval( $order_id ) ];
+		$wp_query->query_vars = [ 'order-pay' => strval( $order_id ) ];
+
+		update_option( '_wcpay_feature_tokenized_cart_ece', '1' );
+		add_filter( 'woocommerce_is_checkout', '__return_true' );
+
+		$this->assertTrue( $this->system_under_test->should_show_express_checkout_button() );
+
+		remove_filter( 'woocommerce_is_checkout', '__return_true' );
+	}
```
Comment (frosso) [inline tests/unit/express-checkout/test-class-wc-payments-express-checkout-button-helper.php:234]: I attempted to create a test to negate the situation (e.g.: create a test to verify the button isn't rendered)

However, unsetting the billing email via either `$order->set_billing_address( [ 'email' => '' ] )` or `$order->set_billing_email('')` didn't work.
I thought of querying the DB directly, but with HPOS and non-HPOS alternatives, I didn't think it was maintainable.

So here we are 🤷

Comment (mdmoore) [reply tests/unit/express-checkout/test-class-wc-payments-express-checkout-button-helper.php:234]: I tried for a bit on this as well. The email really does not want to update. I'm getting strange results with the `$order->save()` method. The same order ID is returned every time and no logs set within that method are coming through. So I'm with you on this one. Directly modifying the DB seems to be the only way to remove the email address.

**Review by mdmoore — APPROVED** [review]
Tests well and LGTM! 🚢

**Pull request merged**